jsr352/core/src/test/java/org/hibernate/search/jsr352/massindexing/JobParameterValidationListenerTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.jsr352.massindexing;
+
+import org.hibernate.search.exception.SearchException;
+
+import org.junit.Test;
+
+/**
+ * @author Mincong Huang
+ */
+public class JobParameterValidationListenerTest {
```

Comment:
I'll refactor this test after the resolution of ticket [HSEARCH-2632](https://hibernate.atlassian.net/browse/HSEARCH-2632), so that we can introduce the setter method for batch property injection, and make the test more elegant:

```java
public class JobParameterValidationListenerTest {

    private JobParameterValidationListener validationListener;

    @Before
    public void setUp() {
        validationListener = new JobParameterValidationListener();
    }

    @Test(expected = SearchException.class)
    public void validateRowsPerPartition_valueIsNegative() throws Exception {
        validationListener.setCheckpointInterval( 100 );
        validationListener.setRowsPerPartition( -1 );
        validationListener.beforeJob();
    }

    ...

}
```